modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/Subtask/useSubtasksActions.tsx

```diff
@@ -65,13 +91,45 @@ const useSubtasksActions = () => {
 					title: i18n.translate('users'),
 				}),
 			icon: 'user',
-			name: i18n.translate('assign'),
+			name: ({dueStatus}) => {
+				if (dueStatus.key === SubTaskStatuses.IN_ANALYSIS) {
+					return i18n.translate('assign');
+				}
+
+				if (dueStatus.key === SubTaskStatuses.OPEN) {
+					return i18n.translate('assign-and-begin-analysis');
+				}
+
+				if (dueStatus.key === SubTaskStatuses.COMPLETE) {
+					return i18n.translate('assign-and-reanalyze');
+				}
+			},
 		},
 		{
 			action: (subtask) => completeModal.open(subtask),
+			hidden: ({dueStatus, user}) =>
+				user?.id !== Number(Liferay.ThemeDisplay.getUserId()) ||
+				dueStatus.key !== SubTaskStatuses.IN_ANALYSIS,
 			icon: 'polls',
 			name: i18n.sub('complete-x', ''),
 		},
+		{
+			action: (subtask, mutate) => {
+				testraySubTaskImpl.returnToOpen(subtask).then(() => {
+					updateItemFromList(
+						mutate,
+						0,
+						{},
+						{
+							revalidate: true,
+						}
+					);
+				});
+			},
```

Comment:
Suggestion: you can simplify returning directly

```suggestion
			action: (subtask, mutate) => 
				testraySubTaskImpl.returnToOpen(subtask).then(() => 
					updateItemFromList(
						mutate,
						0,
						{},
						{
							revalidate: true,
						}
					);
				),
			
```